sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1926,6 +1926,23 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val MAX_COMMON_EXPRS_IN_COLLAPSE_PROJECT =
+    buildConf("spark.sql.optimizer.maxCommonExprsInCollapseProject")
```

Comment:
If we set this value to 1, all the existing tests can pass?

Reply:
I guess not. We might have at lease few common expressions in collapsed projection. If set to 1, any duplicated expression is not allowed.